config.go

```diff
@@ -323,6 +325,37 @@ func golang(c *Config) {
 	}
 }
 
+// java config.
+func java(c *Config) {
+	if c.Hooks.Build.IsEmpty() {
+		// attempt to handle either Gradle or Maven builds and their respective wrappers
+		// assumes build/package results in shaded jar named server.jar
+		if util.Exists("pom.xml") {
+			if util.Exists("mvnw") {
+				c.Hooks.Build = config.Hook{`./mvnw clean package && cp target/server.jar .`}
+			} else {
+				c.Hooks.Build = config.Hook{`mvn clean package && cp target/server.jar .`}	
+			}				
+		} else if util.Exists("build.gradle") {
+			if util.Exists("gradlew") {
+				c.Hooks.Build = config.Hook{`./gradlew clean build && cp build/libs/server.jar .`}
+			} else {
+				c.Hooks.Build = config.Hook{`gradle clean build && cp build/libs/server.jar .`}	
+			}			
+		} else {
+			return
+		}
+	}
+
+	if c.Hooks.Clean.IsEmpty() {
+		c.Hooks.Clean = config.Hook{`rm server.jar`}
```

Comment:
Do you want to run `mvn clean` or `gradle clean` here too?

Reply:
My understanding based on looking at other runtime implementations is that this is only concerned with removing the server artifact. Is my understanding wrong? In any case, `mvn clean` or `gradle clean` is already called during build commands so not sure what it buys us adding them here too.

Reply:
On some other platforms, the server artifact is the _only_ artifact.

Maven / gradle will produce a `build/` or `target/` directory.  That directory needs to be gitignore'd (and implicitly upignore'd) or they will be included in the zipfile uploaded for that.

(run `up build` to see what I mean).

If do you not clean the `build/` or `target/` directories (by running `mvn clean`) then up will leave those build artifacts around your function's directory.

Since this is defining the _default_ behavior, I think up by default should leave your source tree clean after a successful deploy, not with random build / target directories sitting around.

Reply:
Understood about the build/target directories around. In my tests and up-examples I've been careful to include those in `.upignore`. 

My thinking is that the directories (and any other artifacts that shouldn't be packaged via up should be left alone as there might be additional steps in a CI pipeline, etc. that may depend on them after `up` operations. It seems that if we want to "clean" certain files/directories then there should be a notion of a default `.upignore`. Here's an example of my `.upignore` (which includes many other files and directories besides `build/`):
```
.DS_Store
.gradle
gradle
.idea
build
*.gradle
src
gradlew*
*.iml
up.json
.upignore
*.md
out.zip
```
For now, I'll go ahead and add `clean` to `c.Hooks.Clean`.